tests/integration/suite/daprd/resiliency/apps/noretryfilter.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apps
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(noretryfilter))
+}
+
+type noretryfilter struct {
+	daprd1   *daprd.Daprd
+	daprd2   *daprd.Daprd
+	counters sync.Map
+}
+
+func (d *noretryfilter) Setup(t *testing.T) []framework.Option {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(r.Method))
+	})
+
+	handler.HandleFunc("/retry", func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		key := r.URL.Query().Get("key")
+		if key == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		c, _ := d.counters.LoadOrStore(key, &atomic.Int32{})
+		counter := c.(*atomic.Int32)
+		counter.Add(1)
+
+		respStatusCode, err := strconv.Atoi(string(body))
+		if (err != nil) || (respStatusCode < 100) || (respStatusCode >= 600) {
+			// Trying to write a bad status code will cause app to crash.
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		w.Header().Set("content-type", "text/plain")
+		w.WriteHeader(respStatusCode)
+		w.Write([]byte{})
+	})
+
+	srv := prochttp.New(t, prochttp.WithHandler(handler))
+
+	resiliency := `
+apiVersion: dapr.io/v1alpha1
+kind: Resiliency
+metadata:
+  name: myresiliency
+spec:
+  policies:
+    retries:
+      DefaultAppRetryPolicy:
+        policy: constant
+        duration: 100ms
+        maxRetries: 3
+        noRetryOnCodes:
+          - "505"
+          - "2"
+          - "invalid"
+          - "501-503,509"
+          - "546,547"
```

Comment:
What happens when this is defined along with `retryOnCodes`, and what happens when they overlap?

Reply:
There is an error and the policy is invalid.

Reply:
When there is overlap, can there be a convention like "the new entry overrides the old one"...so the rule is still valid and not ignored silently